dbms/src/Flash/Planner/plans/PhysicalAggregation.cpp

```diff
@@ -102,9 +104,21 @@ void PhysicalAggregation::transformImpl(DAGPipeline & pipeline, Context & contex
         aggregate_descriptions,
         is_final_agg);
 
-    /// If there are several sources, then we perform parallel aggregation
-    if (pipeline.streams.size() > 1 || pipeline.streams_with_non_joined_data.size() > 1)
+    if (fine_grained_shuffle.enable())
     {
+        /// For fine_grained_shuffle, just do aggregation in streams independently
```

Comment:
check `pipeline.streams_with_non_joined_data.empty()` here

Reply:
I'll add a RUNTIME check here, it would be ensured in TiDB side, since I just want to support seprate join and agg, no combination of them.

Reply:
Done.